core/domain/activity_domain.py

```diff
@@ -28,17 +28,38 @@ class ActivityReference(object):
     """
 
     def __init__(self, activity_type, activity_id):
+        """Constructs an ActivityReference instance.
+
+        Args:
+          activity_type: str.
+          activity_id:   str.
+        """
         self.type = activity_type
         self.id = activity_id
 
     def get_hash(self):
+        """Gets instance hash.
+
+        Returns:
+          A str in the form of 'type:id'.
+        """
         return '%s:%s' % (self.type, self.id)
 
     def validate(self):
+        """Checks if instance type is valid.
```

Comment:
Checks that all fields of the instance are valid.

(At least, this is what it's meant to do, conceptually. We don't have validation for activity id because those can be arbitrary strings.)